packages/vite/src/node/server/index.ts

```diff
@@ -120,6 +120,20 @@ export interface ServerOptions {
    * Should start and end with the `/` character
    */
   base?: string
+  /**
+   * Options for files served via '/\@fs/'.
+   */
+  fsServe?: FileSystemServeOptions
+}
+
+export interface FileSystemServeOptions {
+  /**
+   * Restrict accessing files outside this directory will result in a 403.
+   *
+   * Accepts absolute path or a path relative to project root.
+   * Will try to search up for workspace root by default.
+   */
+  root?: string
```

Comment:
As discussion with @patak-js, we are likely to add more options to configure `/@fs/` to enhance  further security in the future. For example, disabling dotfiles (`.env`) by default (e.g. [sirv](https://github.com/lukeed/sirv/tree/master/packages/sirv#optsdotfiles)). Changed this to an object for further extensions.